src/main/java/org/folio/service/budget/BudgetService.java

```diff
@@ -60,146 +59,109 @@ public void deleteById(String id, Context vertxContext, Map<String, String> head
     logger.debug("deleteById:: Trying to delete finance storage budgets with id {}", id);
     vertxContext.runOnContext(v -> {
       DBClient client = new DBClient(vertxContext, headers);
-      budgetDAO.getBudgetById(id, client)
-        .compose(budget -> checkTransactions(budget, client).map(budget))
-        .compose(budget -> client.startTx()
-          .compose(t -> unlinkGroupFundFiscalYears(id, client))
-          .compose(t -> budgetDAO.deleteBudget(id, client))
-          .compose(t -> deleteAllocationTransactions(budget, client))
-          .compose(t -> client.endTx())
-          .onComplete(reply -> {
-            if (reply.failed()) {
-              logger.error("deleteById:: Deleting finance storage budgets with id {} failed", id, reply.cause());
-              client.rollbackTransaction();
-            }
-          }))
-        .onComplete(handleNoContentResponse(asyncResultHandler, id, "deleteById:: Budget {} {} deleted"));
+      client.withTrans(conn -> budgetDAO.getBudgetById(id, conn)
+        .compose(budget -> checkTransactions(budget, conn).map(budget)
+          .compose(t -> unlinkGroupFundFiscalYears(id, conn))
+          .compose(t -> budgetDAO.deleteBudget(id, conn))
+          .compose(t -> deleteAllocationTransactions(budget, conn))
+          .onFailure(e -> logger.error("deleteById:: Deleting finance storage budgets with id {} failed", id, e))
+        ).onComplete(handleNoContentResponse(asyncResultHandler, id, "deleteById:: Budget {} {} deleted"))
```

Comment:
.onComplete should be placed after closing last ')' for withTrans method , because now we are returning result when transaction is still in progress and can be rolled back

Reply:
done